docs: add "Modifying the Head Element" example (#1226)
denoland/fresh: docs/examples/modifying-the-head-element.md

> # Modifying Web Page Metadata via the ```<Head /> ```Element

Reviewer: ```suggestion
# Modifying Web Page Metadata via the `<Head />` Element
```
Triple backticks aren't needed for inline code snippets. Can you please use a single backtick where appropriate?

# Modifying the `<head>` Element

We can use the `<Head />` component in `$fresh/runtime.ts` to modify the
`<head>` element. By adding changes to the inner components of `<Head />` we can

> add a new components to the `<head />` element of the display page.

Reviewer: ```suggestion
add a new components to the `<head>` element of the display page.
```

> ## Common components for modifying `<head>`
> 
> - `<title>` Setting the document title
> - `<meta>` Specifying metadata
> - `<link>` Link external resources
> - `<script>` reference JavaScript code

Reviewer: ```suggestion
## Common uses for modifying `<head>`

- Setting the document title using `<title>`
- Specifying page metadata using `<meta>`
- Linking to external resources using `<link>`
- Including JavaScript code via `<script>`
```

# Modifying the `<head>` Element

We can use the `<Head />` component in `$fresh/runtime.ts` to modify the
`<head>` element. By adding changes to the inner components of `<Head />` we can
add a new components to the `<head />` element of the display page.

## Common components for modifying `<head>`

- `<title>` Setting the document title
- `<meta>` Specifying metadata
- `<link>` Link external resources
- `<script>` reference JavaScript code

```tsx
// routes/index.tsx
import { Head } from "$fresh/runtime.ts";
import Counter from "../islands/Counter.tsx";

export default function Home() {
  return (
    <>
      <Head>
        <meta charset="UTF-8" />
        <title>Fresh App</title>
        <meta
          name="description"
          content="This is a brief description of Fresh"
        />
        <link rel="stylesheet" href="styles.css" />
        <script src="script.js"></script>
      </Head>
      <div class="p-4 mx-auto max-w-screen-md">

>         <img
>           src="/logo.svg"
>           class="w-32 h-32"
>           alt="the fresh logo: a sliced lemon dripping with juice"
>         />
>         <p class="my-6">
>           Welcome to `fresh`. Try updating this message in the
>           ./routes/index.tsx file, and refresh.
>         </p>
>         <Counter start={3} />

Reviewer 2: ```suggestion
        <h1>Hello World</h1>
```

This part of the example is not important to what we are trying to bring across, so we can simplify. This allows users to focus on the important bit (the `<Head>` tag).
